sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/lang/SqliteHighlighter.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.intellij.lang
+
+import com.alecstrong.sqlite.psi.core.SqliteLexerAdapter
+import com.alecstrong.sqlite.psi.core.psi.SqliteTypes
+import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
+import com.intellij.openapi.editor.colors.TextAttributesKey
+import com.intellij.openapi.fileTypes.SyntaxHighlighterBase
+import com.intellij.psi.tree.IElementType
+
+class SqliteHighlighter : SyntaxHighlighterBase() {
+  override fun getHighlightingLexer() = SqliteLexerAdapter()
+
+  override fun getTokenHighlights(tokenType: IElementType) =
+    SyntaxHighlighterBase.pack(textAttributesKey[tokenType.index])
+
+  companion object {
+    private val SQLITE_KEYWORD = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.KEYWORD", DefaultLanguageHighlighterColors.KEYWORD
+    )
+    private val SQLITE_NUMBER = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.NUMBER", DefaultLanguageHighlighterColors.NUMBER
+    )
+    private val SQLITE_STRING = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.STRING", DefaultLanguageHighlighterColors.STRING
+    )
+    private val SQLITE_OPERATOR = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.OPERATOR", DefaultLanguageHighlighterColors.OPERATION_SIGN
+    )
+    private val SQLITE_PAREN = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.PAREN", DefaultLanguageHighlighterColors.PARENTHESES
+    )
+    private val SQLITE_DOT = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.DOT", DefaultLanguageHighlighterColors.DOT
+    )
+    private val SQLITE_SEMICOLON = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.SEMICOLON", DefaultLanguageHighlighterColors.SEMICOLON
+    )
+    private val SQLITE_COMMA = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.COMMA", DefaultLanguageHighlighterColors.COMMA
+    )
+    private val SQLITE_DOC = TextAttributesKey.createTextAttributesKey(
+        "SQLITE.DOC", DefaultLanguageHighlighterColors.DOC_COMMENT
+    )
+
+    private val textAttributesKey = mutableMapOf<Short, TextAttributesKey>()
```

Comment:
Is this actually mutable at runtime? How about `mapOf(a to b, c to d)`-style initialization?

Reply:
Agree, that'll be less noisy, I'll change it